python/cudf/cudf/core/dataframe.py

```diff
@@ -1287,40 +1289,28 @@ def at(self):
     def columns(self):
         """Returns a tuple of columns
         """
-        if hasattr(self, "multi_cols"):
-            return self.multi_cols
-        else:
-            name = self._columns_name
-            return pd.Index(self._data.keys(), name=name, tupleize_cols=False)
+        return pd.Index(
+            self._data.names,
+            name=self._data.name,
+            tupleize_cols=self._data.multiindex,
+        )
 
     @columns.setter
     def columns(self, columns):
-        if isinstance(columns, cudf.MultiIndex):
-            if len(columns) != len(self.columns):
-                msg = (
-                    f"Length mismatch: Expected axis has %d elements, "
-                    "new values have %d elements"
-                    % (len(self.columns), len(columns))
-                )
-                raise ValueError(msg)
-            """
-            new_names = []
-            for idx, name in enumerate(columns):
-                new_names.append(name)
-            self._rename_columns(new_names)
-            """
-            self.multi_cols = columns
-        elif isinstance(columns, pd.MultiIndex):
-            self.multi_cols = cudf.MultiIndex.from_pandas(columns)
-        else:
-            if hasattr(self, "multi_cols"):
-                delattr(self, "multi_cols")
-            self._rename_columns(columns)
-            if hasattr(columns, "name"):
-                self._columns_name = columns.name
+        if isinstance(columns, (cudf.MultiIndex, cudf.Index)):
+            columns = columns.to_pandas()
+        if columns is None:
+            columns = pd.Index(range(len(self._data.columns)))
```

Comment:
Instead of constructing a Pandas index here we can likely just use the range as it passes into a zip below.